sessionize/lib/src/main/java/co/touchlab/sessionize/platform/FunctionsJVM.kt

```diff
@@ -75,3 +89,93 @@ actual fun logException(t: Throwable) {
 actual fun settingsFactory(): Settings.Factory = PlatformSettings.Factory(AndroidAppContext.app)
 
 actual fun createUuid(): String = UUID.randomUUID().toString()
+
+val notificationPublisher: BroadcastReceiver = NotificationPublisher()
+
+actual fun createLocalNotification(title:String, message:String, timeInMS:Long, notificationId: Int) {
+
+    var notificationTime = timeInMS - tenMinutesInMS
+    if(notificationTime < co.touchlab.sessionize.platform.Date(Date()).toLongMillis()){
+        notificationTime = co.touchlab.sessionize.platform.Date(Date()).toLongMillis()
+    }
+
+    // Building Notification
+    val channelId = AndroidAppContext.app.getString(R.string.notification_channel_id)
+    var builder = NotificationCompat.Builder(AndroidAppContext.app, channelId)
+            .setSmallIcon(R.drawable.notification_tile_bg)
+            .setContentTitle(title)
+            .setContentText(message)
+            .setWhen(notificationTime)
+            .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+
+
+    // Building Intent wrapper
+    val intent = Intent().also { intent ->
+        intent.action = AndroidAppContext.app.getString(R.string.notification_action)
+        intent.putExtra(NotificationPublisher.NOTIFICATION_ID, notificationId)
+        intent.putExtra(NotificationPublisher.NOTIFICATION, builder.build())
+    }
+    val pendingIntent = PendingIntent.getBroadcast(AndroidAppContext.app,notificationId, intent, PendingIntent.FLAG_CANCEL_CURRENT)
+
+    // Scheduling Intent
+    val alarmManager = AndroidAppContext.app.getSystemService(Context.ALARM_SERVICE) as AlarmManager
+    alarmManager.set(AlarmManager.RTC_WAKEUP, notificationTime, pendingIntent)
+
+}
+
+class NotificationPublisher : BroadcastReceiver() {
+
+    override fun onReceive(context: Context, intent: Intent) {
+        val notification = intent.getParcelableExtra<Notification>(NOTIFICATION)
+        val notificationId = intent.getIntExtra(NOTIFICATION_ID, 0)
+
+        with(NotificationManagerCompat.from(AndroidAppContext.app)) {
+            // notificationId is a unique int for each notification that you must define
+            this.notify(notificationId, notification)
+        }
+    }
+
+    companion object {
+        var NOTIFICATION_ID = "notification_id"
+        var NOTIFICATION = "notification"
+    }
+}
+
+actual fun cancelLocalNotification(notificationId: Int){
+    with(NotificationManagerCompat.from(AndroidAppContext.app)) {
+        this.cancel(notificationId)
+    }
+}
+
+actual fun initializeNotifications(){
+    val filter = IntentFilter(AndroidAppContext.app.getString(R.string.notification_action))
+    AndroidAppContext.app.registerReceiver(notificationPublisher, filter)
+
+    createNotificationChannel()
+}
+
+actual fun deinitializeNotifications(){
```

Comment:
If we unregister the BroadcastReceiver when you leave the app, doesn't this mean we won't get a notification about the session when the alarm triggers?

Reply:
Oh good point, that is true.

Reply:
I updated it to work after the application is closed. Tested in iOS and Android